pkg/ingester/ingester.go

```diff
@@ -501,28 +503,38 @@ func (i *Ingester) updateActiveSeries(now time.Time) {
 	}
 }
 
-// Go through all tenants and apply the current max-exemplars setting.
-// If it changed, tsdb will resize the buffer; if it didn't change tsdb will return quickly.
-func (i *Ingester) applyExemplarsSettings() {
+// applyTSDBSettings goes through all tenants and applies
+// * The current max-exemplars setting. If it changed, tsdb will resize the buffer; if it didn't change tsdb will return quickly.
+// * The current out of order allowance. If it changes from 0 to >0, then a new Write-Behind-Log gets created for that tenant.
+func (i *Ingester) applyTSDBSettings() {
 	for _, userID := range i.getTSDBUsers() {
 		globalValue := i.limits.MaxGlobalExemplarsPerUser(userID)
 		localValue := i.limiter.convertGlobalToLocalLimit(userID, globalValue)
-		// We populate a Config struct with just one value, which is OK
-		// because Head.ApplyConfig only looks at one value.
+
+		// OutOfOrderAllowance is a Duration only for convenience. TSDB will use OutOfOrderAllowance.Milliseconds()
+		// as the final value for the allowance which should match our desired value for the unit of the timestamp.
+		// Since we use milliseconds for our timestamps, we can directly use the duration without any modifications.
+		oooAllowance := i.limits.OutOfOrderAllowance(userID)
```

Comment:
Shall we normalize values less than 0 to 0?

Reply:
That is being taken care inside TSDB. But I will do it here as well.

Reply:
Fixed in https://github.com/grafana/mimir/pull/2139/commits/6c3c72c28a55c2b80aaa62700e235acc94d68841